### PR TITLE
Implement version-scoped column field query discovery support

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR introduces a /fields and /discover endpoint within the /form/version part of the API which allows users to laterally slice and query for certain subsets of information from their form submissions. This is needed in order to eventually allow for customizable submission columns on the frontend.

[SHOWCASE-2180](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-2180)

## Types of changes

New feature (non-breaking change which adds functionality)
Documentation (non-breaking change with enhancements to documentation)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

Check the OpenAPI spec for latest details on endpoint changes. Unit tests coming up soon.
Full path endpoints being proposed:
- `GET {{hostname}}/app/api/v1/forms/:formId/versions/:versionId/fields`
  - https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-220/api/v1/docs#operation/readVersionFields
- `GET {{hostname}}/app/api/v1/forms/:formId/versions/:versionId/submissions/discover?fields=foo,bar,baz`
  - https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-220/api/v1/docs#operation/listDiscoverSubmissions

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-220

File: app/src/forms/form/service.js [hunk 1]
```diff
@@ -307,7 +325,14 @@ const service = {
     }
   },
 
-  readSubmission: async (id) => {
+  listSubmissionFields: (formVersionId, fields) => {
```
Comment (TimCsaky) [inline app/src/forms/form/service.js:328]: don't you need to flatten the array of returned objects that you got for each submission into one list of unique fields?

Comment (TimCsaky) [reply app/src/forms/form/service.js:328]: ok. maybe you already did that

**Review by TimCsaky — APPROVED** [review]
nice!

**Review by loneil — APPROVED** [review]

**Pull request merged**